src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts

```diff
@@ -67,22 +67,41 @@ export class GithubAppsLogsComponent implements OnInit {
     this.loading = true;
     this.dataSource.sort = this.sort;
     this.dataSource.paginator = this.paginator;
-    this.lambdaEventsService
-      .getLambdaEventsByOrganization(this.matDialogData)
-      .pipe(
-        finalize(() => {
-          this.loading = false;
-          this.updateContentToShow(this.lambdaEvents);
-        })
-      )
-      .subscribe(
-        (lambdaEvents) => (this.lambdaEvents = lambdaEvents),
-        (error) => {
-          this.lambdaEvents = null;
-          const detailedErrorMessage = AlertService.getDetailedErrorMessage(error);
-          this.matSnackBar.open(detailedErrorMessage);
-        }
-      );
+    if (this.matDialogData.getUserEvents) {
+      this.lambdaEventsService
+        .getUserLambdaEvents(this.matDialogData.value)
+        .pipe(
```

Comment:
The blocks 73-86 and 90-103 are identical, I think. You should be able to remove the duplication, e.g., 

```
const lambdaEvents = this.matDialogData.getUserEvents ? this.lambdaEventsService
        .getUserLambdaEvents(this.matDialogData.value) :  this.lambdaEventsService
        .getLambdaEventsByOrganization(this.matDialogData.value); // Or have this logic in a method
lambdaEvents.pipe(
...
```